enos/enos-scenario-agent.hcl

```diff
@@ -3,19 +3,83 @@
+    or in your 'enos-local.vars' file:
+
+    Required variables
+      - aws_ssh_private_key_path (more info about AWS SSH keypairs: https://eng-handbook.hashicorp.services/internal-tools/enos/getting-started/#set-your-aws-key-pair-name-and-private-key)
+      - aws_ssh_keypair_name
+      - vault_build_date
+      - vault_product_version
+      - vault_revision
+
+    Optional variables
+      - aws_region (if different from the default value in enos-variables.hcl)
+      - consul_license_path (if using an ENT edition of Consul)
+      - distro_version_<distro> (if different from the default version for your target
+      distro; see list of supported distros and versions in enos-globals.hcl)
+      - vault_license_path (if using an ENT edition of Vault)
+    
+    4. Choose one of the following ways to get a Vault artifact and set the appropriate variables:
+
+      a. 'artifact_source:crt'
+      This will use a Vault artifact that you have already downloaded. Set the following variable:
+
+      ENOS_VAR_vault_artifact_path=path/to/existing-vault-artifact.[zip | rpm | deb]
+
+      b. 'artifact_source:local'
+      This will build a Vault .zip bundle from your local branch. Set the following variable:
+
+      ENOS_VAR_vault_artifact_path=path/to/where/vault-should-be-built.zip
+
+      c. 'artifact_source:artifactory'
+      This will download a Vault artifact of the specified version, edition, and revision SHA
+      from the 'stable' repo in Artifactory. Set the following variables:
+
+      export ENOS_VAR_artifactory_username=your-user
+      export ENOS_VAR_artifactory_token=your-token
+    
+    5. Choose the matrix variants you want to use, and launch the scenario with the appropriate
+    filter for those variants, e.g.:
+
+    $ enos scenario launch agent arch:amd64 artifact_source:crt artifact_type:bundle backend:consul config_mode:env consul_edition:ent consul_version:1.16.3 distro:ubuntu edition:ent.hsm.fips1402 seal:pkcs11
+
+    Notes:
+    - Enos will run all matrix variant combinations that match your filter. If you specify one
+      variant for each matrix item, the filter will produce and run only one scenario. Even if you are
+      using a Raft backend, you may want to specify a consul_version (though it functionally will not
+      do anything since you're not using Consul) so that Enos does not run multiple scenarios (one for
+      each Consul version in the matrix).
+
+    - If you want to use the 'distro:leap' variant you must first accept SUSE's terms for the AWS
+      account. To verify that your account has agreed, sign-in to your AWS through Doormat,
+      and visit the following links to verify your subscription or subscribe:
+        arm64 AMI: https://aws.amazon.com/marketplace/server/procurement?productId=a516e959-df54-4035-bb1a-63599b7a6df9
+        amd64 AMI: https://aws.amazon.com/marketplace/server/procurement?productId=5535c495-72d4-4355-b169-54ffa874f849
```

Comment:
Nit: turn these into links.

Reply:
This doesn't get put in markdown at any point, it's just a blurb of HCL.